server/src/main/java/org/apache/cloudstack/diagnostics/DiagnosticsServiceImpl.java

```diff
@@ -126,10 +188,333 @@ protected String prepareShellCmd(String cmdType, String ipAddress, String option
         }
     }
 
+    @Override
+    @ActionEvent(eventType = EventTypes.EVENT_SYSTEM_VM_DIAGNOSTICS, eventDescription = "getting diagnostics files on system vm", async = true)
+    public String getDiagnosticsDataCommand(GetDiagnosticsDataCmd cmd) {
```

Comment:
50 line method, factor out some code for readibilty?

Reply:
Done.